safeptr.go

```diff
@@ -0,0 +1,32 @@
+//go:build go1.18
+// +build go1.18
+
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package klog
+
+// SafePtr accepts a pointer of any type (T) as an argument and returns the
+// same pointer if it's not nil, otherwise it returns nil.
+// This function is useful for safely handling pointers and preventing nil pointer dereferencing when the type
+//
+// a) implements interfaces called by the logger like `fmt.Stringer` and b) those implementations do not check for nil themselves.
+func SafePtr[T any](p *T) any {
```

Comment:
Can you re-wrap the comment?

Please also squash everything into one commit and update the PR description (still mentions Go 1.21).